service/controller/resource/promxy/types.go

```diff
@@ -54,13 +54,15 @@ func (p *PromxyConfig) add(group *ServerGroup) {
 }
 
 func (p *PromxyConfig) remove(group *ServerGroup) {
-	var index int
-	for key, val := range p.ServerGroups {
-		if val.PathPrefix == group.PathPrefix {
-			index = key
+	var s []*ServerGroup
```

Comment:
This makes it easier to read for sure :D For my curiosity, was there any issue with the existing code?

Reply:
Existing code was correct, but I thought this would clean up the configmap automatically. After testing I realized this won't work because resources might be gone. So we have to clean up manually.

Reply:
Thanks :)